modules/services/ollama.nix

```diff
@@ -0,0 +1,116 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.ollama;
+
+  ollamaPackage = if cfg.acceleration == null then
+    cfg.package
+  else
+    cfg.package.override { inherit (cfg) acceleration; };
+
+  postStartScript = pkgs.writeShellScript "ollama-post-start" ''
+    set -x
+    export OLLAMA_HOST=${escapeShellArg cfg.host}:${builtins.toString cfg.port}
+    for model in ${escapeShellArgs cfg.loadModels}
+    do
+      ${escapeShellArg (getExe ollamaPackage)} pull "$model"
+    done
```

Comment:
why is it postStart ? does it need the server running first to pull images ?

Reply:
Yes, it does:

```console
ollama pull llama3.3:70b
Error: could not connect to ollama app, is it running?
```

Reply:
This is following the convention from the NixOS module. I guess we could also drop the model pull support. In my experience it didn't work very well. Especially if you don't have the models already, the initial pull can be slow.

I see now that they have migrated to a separate service (likely due to the issues described), we should probably also do that:
https://github.com/NixOS/nixpkgs/blob/a04634d3e7409aeba2dc2cd53ffc678250cf0ae3/nixos/modules/services/misc/ollama.nix#L266

Reply:
If you are not convinced by it (I am not either, makes startup impredictible), let's pull it out and we can add this later if anyone expresses a need for it. The feature might be more necessary in a system-level declarative setup but I guess here the user can just run ollama pull by himself (providing the good OLLAMA_HOST) and if you install via home-manager, it's for personal local use.

Reply:
@teto module updated to drop this feature for now, and we can add it properly later, perhaps inspired by what was done in the NixOS module.